rfcs/1-2FA-email/rfc.md

```diff
@@ -0,0 +1,52 @@
+# Summary
+
+Enable two factor authentication via email. The user would provide their email/username and password (existing functionality) then be prompted to input a 6 digit code delivered to their email (new functionality) in order to login. The 6 digit code will be generated as an TOTP (Time based One Time Password) code.
+
+### Benefit
+
+Many companies are now requiring 2FA on all accounts and platforms their employees may use. The addition of this feature would allow companies using oxidauth to comply with this requirement.
+
+
+### Technical Flow
+Each user is assigned a static auth key at creation (stored in new table auth_keys). From the frontend login screen, a user provides username & password combo, if password is correct:
+
+- a temporary jwt is sent to the browser, allowing the user to access the 2FA code input screen
+- a 6 digit code is created from the secret key + current time, then emailed to the user
+- user inputs the 6 digit code from their email which is diffed against the secrety key + current time
+- if 5 minutes have passed, the new code will not match the user entered code, and login will fail. Returns an error.
+- if user input code does match the new code for any reason, login will fail. Returns an error
+- if user requests a new code, the 6 digit code is recreated with new current time, and emailed to the user
```

Comment:
I think we need another configuration here where you can't request a new one until the existing one has expired? Or maybe this is a separate kind of configuration where you decide the rate that codes can be generated.